template/{{app_name}}/app/helpers/uswds_form_builder.rb

```diff
@@ -59,49 +72,75 @@ def radio_button(attribute, tag_value, options = {})
   end
 
   def select(attribute, choices, options = {}, html_options = {})
-    classes = "usa-select"
-
-    html_options[:class] ||= ""
-    html_options[:class].prepend("#{classes} ")
+    append_to_option(html_options, :class, " usa-select")
 
     label_text = options.delete(:label)
 
-    us_form_group(attribute: attribute) do
+    form_group(attribute) do
       us_text_field_label(attribute, label_text, options) + super(attribute, choices, options, html_options)
     end
   end
 
   def submit(value = nil, options = {})
-    options[:class] ||= ""
-    options[:class].prepend("usa-button ")
-
-    super(value, options)
-  end
-
-  def honeypot_field
-    spam_trap_classes = "opacity-0 position-absolute z-bottom top-0 left-0 height-0 width-0"
-    label_text = "Do not fill in this field. It is an anti-spam measure."
+    append_to_option(options, :class, " usa-button")
 
-    @template.content_tag(:div, class: "usa-form-group #{spam_trap_classes}") do
-      label(:spam_trap, label_text, { tabindex: -1, class: "usa-label #{spam_trap_classes}" }) +
-      @template.text_field(@object_name, :spam_trap, { autocomplete: "false", tabindex: -1, class: "usa-input #{spam_trap_classes}" })
+    if options[:big]
+      append_to_option(options, :class, " usa-button--big margin-y-6")
     end
+
+    super(value, options)
   end
 
   ########################################
   # Custom helpers
   ########################################
 
+  def tax_id_field(attribute, options = {})
+    options[:inputmode] = "numeric"
+    options[:placeholder] = "_________"
+    options[:width] = "md"
+
+    # Actual USWDS mask functionality broken until this is fixed:
+    # https://github.com/uswds/uswds/issues/5517
+    # append_to_option(options, :class, " usa-masked")
```

Comment:
This upstream issue has been marked as resolved since [v3.8.1](https://github.com/uswds/uswds/releases/tag/v3.8.1), have we tested it?

The lock file looks be using USWDS 3.8.0 at the moment, so a minor bump to resolve this would be okay IMO, we don't need to do a big update to 3.11.0 as a part of this work.

Also the content `us_form_with.tax_id_format` needs added, like for the date picker below.

Though big picture, not sure about including a "tax_id_field" as a part of the generic Rails app. :thinking: @lorenyu any thoughts?

Reply:
I don't have strong preferences either way at this point since I think it's relatively easy to clean up one way or another later on, so I'm biased towards doing whatever is simplest for now. Longer term I could see this being part of flex.